internal/registry/registry.go

```diff
@@ -119,25 +118,7 @@ func (r *registry) getPullOptions(ctx context.Context, image string, tlsOptions
 		return nil, fmt.Errorf("image url %s is not valid, does not contain hash or tag", image)
 	}
 
-	options := []crane.Option{
-		crane.WithContext(ctx),
-	}
-
-	if tlsOptions != nil {
-		if tlsOptions.Insecure {
-			options = append(options, crane.Insecure)
-		}
-
-		if tlsOptions.InsecureSkipTLSVerify {
-			rt := http.DefaultTransport.(*http.Transport).Clone()
-			rt.TLSClientConfig.InsecureSkipVerify = true
-
-			options = append(
-				options,
-				crane.WithTransport(rt),
-			)
-		}
-	}
+	options := r.getTransportOptions(tlsOptions.Insecure, tlsOptions.InsecureSkipTLSVerify)
```

Comment:
tlsOptions can be nil, no? so calling this function without verifying will cause a panic

Reply:
Good spot. I have reverted all the changes to this function.

I was trying to reduce the duplicated code but my testing is all focused on the signing image and doesn't call this function at all so I missed the risk of panic here. I still think we should replace this code with a call to getTransportOptions() but it is greatly expanding the surface area of this change into stuff like preflight so it probably needs to be in a commit of its own that can be tested separately.